pkg/updater/service/apm_inject.go

```diff
@@ -0,0 +1,289 @@
+	}
+
+	// Remove the launcher preload path from the file
+	ldSoPreload = bytes.Replace(ldSoPreload, []byte(launcherPreloadPath+"\n"), []byte{}, -1)
+
+	perms := os.FileMode(0644)
+	if stat != nil {
+		perms = stat.Mode()
+	}
+	err = os.WriteFile("/tmp/ld.so.preload.tmp", ldSoPreload, perms)
+	if err != nil {
+		return err
+	}
+
+	return executeCommand(string(replaceLDPreloadCommand))
+}
+
+// setAgentConfig adds the agent configuration for the APM injector if it is not there already
+// We assume that the agent file has been created by the installer's postinst script
+//
+// Note: This is not safe, as it assumes there were no changes to the agent configuration made without
+// restart by the user. This means that the agent can crash on restart. This is a limitation of the current
+// installer system and this will be replaced by a proper experiment when available. This is a temporary
+// solution to allow the APM injector to be installed, and if the agent crashes, we try to detect it and
+// restore the previous configuration
+func (a *apmInjectorInstaller) setAgentConfig() (err error) {
+	runPath := path.Join(a.installPath, "inject", "run")
+	apmSocketPath := path.Join(runPath, "apm.socket")
+	dsdSocketPath := path.Join(runPath, "dsd.socket")
+
+	err = backupAgentConfig()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			restoreErr := restoreAgentConfig()
+			if restoreErr != nil {
+				log.Warnf("Failed to restore agent config: %v", restoreErr)
+			}
+		}
+	}()
+
+	content, err := os.ReadFile(datadogConfigPath)
+	if err != nil {
+		return err
+	}
+
+	if !bytes.Contains(content, injectorConfigPrefix) {
+		content = append(content, injectorConfigPrefix...)
+		content = append(content, []byte(
+			fmt.Sprintf(injectorConfigTemplate, apmSocketPath, dsdSocketPath),
+		)...)
+		content = append(content, injectorConfigSuffix...)
+		err = os.WriteFile(datadogConfigPath, content, 0644)
+		if err != nil {
+			return err
+		}
+	}
+	err = restartTraceAgent()
+	return
+}
+
+// deleteAgentConfig removes the agent configuration for the APM injector
+func (a *apmInjectorInstaller) deleteAgentConfig() (err error) {
+	err = backupAgentConfig()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err != nil {
+			restoreErr := restoreAgentConfig()
+			if restoreErr != nil {
+				log.Warnf("Failed to restore agent config: %v", restoreErr)
+			}
+		}
+	}()
+
+	content, err := os.ReadFile(datadogConfigPath)
+	if err != nil {
+		return err
+	}
+
+	start := bytes.Index(content, injectorConfigPrefix)
+	end := bytes.Index(content, injectorConfigSuffix) + len(injectorConfigSuffix)
+	if start == -1 || end == -1 || start >= end {
+		// Config not found
+		return nil
+	}
+
+	content = append(content[:start], content[end:]...)
+	err = os.WriteFile(datadogConfigPath, content, 0644)
+	if err != nil {
+		return err
+	}
+
+	return restartTraceAgent()
+}
+
+// backupAgentConfig backs up the agent configuration
+func backupAgentConfig() error {
```

Comment:
you can assert that your command is correctly interpreted by the helper with these tests:
https://github.com/DataDog/datadog-agent/blob/cca8e95db5a54fdc3c73b7bed78588a12b922a15/pkg/updater/service/systemd_test.go#L50-L58